plugins/mcreator-core/blockly/js/mcreator_extensions.js

```diff
@@ -375,6 +418,115 @@ function weightedListMutatorMixin(inputType) {
         }, true, ['weight'], true);
 }
 
+function validatingRepeatingInputMixin(mutatorContainer, mutatorInput, inputName, inputProvider, isProperInput = true,

```

Comment:
What if you reconsider **Switch** blocks? (Or that's a lost cause in your opinion?)

Reply:
Switch blocks can be a third-party plugin implementing own JS

Reply:
Having a constant mixin also makes it easier to review the code, as there is no array of arrays of field values, array of field names, etc. to consider

Reply:
By making it constant, I mean something like this:
```js
Blockly.Extensions.registerMutator('procedure_dependencies_mutator',
    {
        mutationToDom: function () {
            // code
        },

        domToMutation: function (xmlElement) {
            // code
        },

        // other mutator functions
    }, undefined, ['procedure_dependencies_mutator_input']);
```
For example, with the current implementation, `fieldValues` is still an array of arrays, even though we know we only have one repeating field.

Reply:
So the current approach using an existing mixin provider for the base and with only one method overridden is worse than entirely rewriting that mixin?

Reply:
The current approach is not as readable and still has "unused" parts (`isProperInput` is always true and can be removed, `fieldValues` can be a single array, there's no need to define a `handleEmptyInput_` function, the input provider can be written explicitly instead of as a parameter etc.). Repeated code isn't always bad.

Reply:
Should I keep the comments?